backend/session/logind.c

```diff
@@ -251,23 +251,17 @@ static bool set_type(struct logind_session *session) {
 		session->path, "org.freedesktop.login1.Session", "SetType",
 		&error, &msg, "s", "wayland");
 	if (ret < 0) {
-		wlr_log(WLR_ERROR, "Failed to set logind session type for session: %s",
+		wlr_log(WLR_DEBUG, "Failed to set logind session type for session: %s",
 			error.message);
 	}
 
 	sd_bus_error_free(&error);
 	sd_bus_message_unref(msg);
 
-	if (ret < 0) {
-		return false;
-	}
-
-	ret = setenv("XDG_SESSION_TYPE", "wayland", 1);
+	setenv("XDG_SESSION_TYPE", "wayland", 1);
```

Comment:
We shouldn't set XDG_SESSION_TYPE if SetType failed, because the two will become out-of-sync in this case.

Reply:
Even though it can't be set on old systemd? They will be out of sync but setting 1/2 seems more correct than 0/2. I'm relaxed though if you want to return early in that case though.

Reply:
Yes, Lennard said keeping both in sync is better.

Reply:
Ok, will update.